api/layer/object.go

```diff
@@ -596,12 +595,18 @@ func (n *layer) initWorkerPool(ctx context.Context, size int, p allObjectParams,
 				wg.Add(1)
 				err = pool.Submit(func() {
 					defer wg.Done()
-					if oi := n.objectInfoFromObjectsCacheOrNeoFS(ctx, p.Bucket, node, p.Prefix, p.Delimiter); oi != nil {
-						select {
-						case <-ctx.Done():
-						case objCh <- oi:
+					oi := n.objectInfoFromObjectsCacheOrNeoFS(ctx, p.Bucket, node, p.Prefix, p.Delimiter)
+					if oi == nil {
+						// try to get object again
```

Comment:
`objectInfoFromObjectsCacheOrNeoFS` can fail `HEAD` request because of network problem. We should try request again before forming partial data.
Also see `3.1` in https://github.com/nspcc-dev/neofs-s3-gw/issues/625